flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSourceSinkFactory.java

```diff
@@ -0,0 +1,228 @@
+			scanOptions = JDBCScanOptions.builder()
+				.setPartitionColumnName(partitionColumnName.get())
+				.setPartitionLowerBound(partitionLower.get())
+				.setPartitionUpperBound(partitionUpper.get())
+				.setNumPartitions(numPartitions.get())
+				.build();
+		} else {
+			scanOptions = null;
+		}
+
+		final Optional<Long> lookupCacheMaxSize =
+			descriptorProperties.getOptionalLong(CONNECTOR_LOOKUP_CACHE_MAX_ENTRIES);
+		final Optional<TimeLength> lookupCacheExpireMs =
+			descriptorProperties.getOptionalTimeLength(CONNECTOR_LOOKUP_CACHE_TTL);
+		final Optional<Integer> lookupMaxRetryTimes =
+			descriptorProperties.getOptionalInt(CONNECTOR_LOOKUP_MAX_RETRIES);
+		JDBCLookupOptions.Builder lookupOptionsBuilder = JDBCLookupOptions.builder();
+		if (lookupCacheMaxSize.isPresent()) {
+			lookupOptionsBuilder = lookupOptionsBuilder.setCacheMaxSize(lookupCacheMaxSize.get());
+		}
+		if (lookupCacheExpireMs.isPresent()) {
+			lookupOptionsBuilder = lookupOptionsBuilder.setCacheExpireMs(lookupCacheExpireMs.get().getMilliseconds());
+		}
+		if (lookupMaxRetryTimes.isPresent()) {
+			lookupOptionsBuilder = lookupOptionsBuilder.setMaxRetryTimes(lookupMaxRetryTimes.get());
+		}
+		final JDBCLookupOptions lookupOptions = lookupOptionsBuilder.build();
+
+		final TableSchema schema = descriptorProperties.getTableSchema(SCHEMA);
+		return JDBCTableSource.builder()
+				.setOptions(options)
+				.setScanOptions(scanOptions)
+				.setLookupOptions(lookupOptions)
+				.setSchema(schema)
+				.build();
+	}
+
+	@Override
+	public StreamTableSink<Tuple2<Boolean, Row>> createStreamTableSink(Map<String, String> properties) {
+		final DescriptorProperties descriptorProperties = getValidatedProperties(properties);
+
+		JDBCOptions options = getJDBCOptions(descriptorProperties);
+		JDBCUpsertTableSink.Builder builder = JDBCUpsertTableSink.builder()
+			.setOptions(options)
+			.setTableSchema(descriptorProperties.getTableSchema(SCHEMA));
+
+		Optional<Integer> flushMaxSize = descriptorProperties.getOptionalInt(CONNECTOR_WRITE_FLUSH_MAX_ENTRIES);
+		Optional<TimeLength> flushIntervalMs = descriptorProperties.getOptionalTimeLength(CONNECTOR_WRITE_FLUSH_INTERVAL);
+		Optional<Integer> maxRetryTimes = descriptorProperties.getOptionalInt(CONNECTOR_WRITE_MAX_RETRIES);
+		if (flushMaxSize.isPresent()) {
+			builder = builder.setFlushMaxSize(flushMaxSize.get());
+		}
+		if (flushIntervalMs.isPresent()) {
+			builder = builder.setFlushIntervalMills(flushIntervalMs.get().getMilliseconds());
+		}
+		if (maxRetryTimes.isPresent()) {
+			builder = builder.setMaxRetryTimes(maxRetryTimes.get());
+		}
+
+		return builder.build();
+	}
+
+	private DescriptorProperties getValidatedProperties(Map<String, String> properties) {
+		final DescriptorProperties descriptorProperties = new DescriptorProperties(true);
+		descriptorProperties.putProperties(properties);
+		new SchemaValidator(true, false, false).validate(descriptorProperties);
```

Comment:
Add a `JDBCValidator` to do validation for JDBC properties, for example, validate optional and required properties.  You can take `ElasticsearchValidator` as an example.